tests/e2e/controllers/messages.js

```diff
@@ -0,0 +1,208 @@
+const assert = require('chai').assert,
+  PouchDB = require('pouchdb'),
+  request = require('request-promise-native'),
+  Url = require('url'),
+  utils = require('../utils');
+
+const adminDb = new PouchDB(process.env.COUCH_URL);
+
+const pendingId = 'ff827d04-7083-4ff3-e53807a03a54d8af';
+
+const data_record = {
+  _id: 'my_data_record',
+  errors: [],
+  form: null,
+  from: '0211111111',
+  reported_date: 1432801258088,
+  tasks: [
+   {
+       state: 'sent',
+       state_history: [
+           {
+               state: 'pending',
+               timestamp: '2015-10-21T07:29:54.493Z'
+           },
+           {
+               state: 'sent',
+               timestamp: '2015-10-21T13:30:21.294Z'
+           }
+       ],
+       messages: [
+           {
+               to: '+254718930649',
+               message: 'Thank you Gareth-the-CHW for registering The Best Patient Ever. Patient ID is 41919. You will receive clinic visit reminders as per child\'s DOB.',
+               uuid: '30956597-0a1c-47a8-9210-56e1f681cae7'
+           }
+       ]
+   }
+  ],
+  read: ['gareth'],
+  kujua_message: true,
+  type: 'data_record',
+  sent_by: 'gareth',
+  scheduled_tasks: [
+    {
+       timestamp: '2017-02-21T12:40:20.616Z',
+       state: 'sent',
+       due: '2017-01-30T12:09:04.137Z',
+       type: 'ReferralHighP',
+       group: 1,
+       state_history: [
+           {
+               timestamp: '2017-01-30T11:54:08.256Z',
+               state: 'scheduled'
+           },
+           {
+               timestamp: '2017-01-30T12:14:00.098Z',
+               state: 'pending'
+           },
+           {
+               timestamp: '2017-02-21T12:40:20.616Z',
+               state: 'sent'
+           }
+       ],
+       messages: [
+           {
+               message: 'Thank you for submitting the Referral Form for Michael Jackson .',
+               to: '+254222222222',
+               uuid: 'ff827d04-7083-4ff3-e53807a03a54cae8'
+           }
+       ]
+   },
+   {
+       timestamp: '2017-01-30T12:19:00.136Z',
+       state: 'pending',
+       due: '2017-01-30T12:14:04.137Z',
+       type: 'ReferralHighP',
+       group: 1,
+       state_history: [
+           {
+               timestamp: '2017-01-30T11:54:08.257Z',
+               state: 'scheduled'
+           },
+           {
+               timestamp: '2017-01-30T12:19:00.136Z',
+               state: 'pending'
+           }
+       ],
+       messages: [
+           {
+               message: 'A vulnerable child has been referred to you. (High Priority)',
+               to: '0222222222',
+               uuid: pendingId
+           }
+       ]
+   }
+  ]
+};
+
+const apiRequest = (endpoint, queryParams) => {
```

Comment:
Isn't this duplicated by utils.request? This is nicer because it uses the library to achieve (I think) the same thing. Maybe improve the utils.request function to use this library too?